SCons/Tool/msvc.py

```diff
@@ -121,8 +124,16 @@ def object_emitter(target, source, env, parent_emitter):
     # https://github.com/SCons/scons/issues/2505
     pch=get_pch_node(env, target, source)
     if pch:
-        if str(target[0]) != SCons.Util.splitext(str(pch))[0] + '.obj':
+        pch_basename = SCons.Util.splitext(str(pch))[0]
+        if str(target[0]) != pch_basename + '.obj':
```

Comment:
This is okay given the pretty constrained context, but maybe ought to use `$OBJSUFFIX` (though then you have to subst is before using...)